### ISSUE TITLE
Pagination component

### ISSUE THREAD

**Issue by buddy-harlow** [Bixal/bixal-site-drupal#249]
Location: https://www.bixal.com/blog
When listening to the pagination component with a screen reader, there is redundant link text being announced. (See NVDA Speech viewer screen shot as follows)

I expect the pagination page links to announce as "Page 1 link" "Page 2 link"

Suggestion
I recommend we adopt [USWDS Pagination component](https://designsystem.digital.gov/components/pagination/)'s visual styling and code as it is more intuitive.

**Comment by buddy-harlow** [comment]
@mpbixal Do you think the closing of #268 will also fix this issue?

**Comment by mattsqd** [comment]
> @mpbixal Do you think the closing of #268 will also fix this issue?

I don't think so, #268 just included the CSS, it did not change the dom. For example here is the USWDS paging component:
![image](https://github.com/user-attachments/assets/3f7d6a58-5ebb-43f7-bebf-9266a81f6fac)
and here is ours:
![image](https://github.com/user-attachments/assets/02f34f59-76ec-4f6b-a111-30cd64f46e02)
Our classes our the same, the difference is that USWDS uses `aria-label` and we use `title`. And the values for those are different.
So I assume, we need to use `aria-label` and switch to just `Page 1`.  Also, check the other `title` values, like 'current page' and 'next page'.

I'm not sure if the `link` in:

> I expect the pagination page links to announce as "Page 1 link" "Page 2 link"

is actually in the `aria-label` as well, or if screen readers just announce the HREF value after they read the `aria-label`.

**Comment by mejiaj** [comment]
@buddy-harlow @mattsqd was the USWDS pagination package added? I'd prefer to grab components 1-by-1 to save on compiled CSS.

When we launched we only had USWDS Core (tokens, functions, mixins).

**Comment by mattsqd** [comment]
@mejiaj Yeah, I believe we turned just on that one component with this PR https://github.com/Bixal/bixal-site-drupal/pull/269/files.
Now we just need to tinker with the dom a little bit, I believe.

**Comment by mejiaj** [comment]
I've created a proof of concept in [`feature/BSD-380-usa-pagination`](https://github.com/Bixal/bixal-site-drupal/tree/feature/BSD-380-usa-pagination) if we want to follow the same behavior for USA Pagination.

Will need some heavy eyes on it though. I tried to mimick the data structure that was in the Bixal pager template, `bixal_uswds/templates/navigation/pager.html.twig`.

## Bixal pagination

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/2e0210a9-8034-4384-80ab-c87e708fd1e0"/>

## Pagination Behavior

Essentially the pagination behavior is like this:

### 1. Show all pages if less than 7

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/f1f6e9df-ba79-47fc-b804-80d62ee92716"/>

### 2. Show first five items of a large dataset

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/11e3d314-635e-4f40-8021-7571b43e8154"/>

### 3. Show last five items of a large dataset

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/2bfcedad-3040-44ba-a35b-ecc7fe65b673"/>

### 4. Default state

By default, show: First page, previous, current, next, last

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/de8aa671-a985-4039-908c-d1dcf67f2030"/>
